### PR TITLE
[Improve][Connector-V2][Iceberg] Using Collections.emptyMap() replaced Collections.EMPTY_MAP

### PR DESCRIPTION
**Pull request by FlechazoW**
## Purpose of this pull request

[Improve] Using Collections.emptyMap() replaced Collections.EMPTY_MAP to avoid 'Unchecked assignment'.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
Hi [FlechazoW](https://github.com/FlechazoW), overall LGTM except few nits. The name of your pull request does not include the name of the module involved. A clear name for the pull request would help make the review easier. You can refer to [Coding Guide](https://seatunnel.apache.org/docs/contribution/coding-guide) get more details about how to submit a high quality pull request.

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/enumerator/IcebergBatchSplitEnumerator.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ public IcebergBatchSplitEnumerator(@NonNull SourceSplitEnumerator.Context<Iceber
                                        @NonNull SourceConfig sourceConfig,
                                        IcebergSplitEnumeratorState restoreState) {
         super(context, sourceConfig, restoreState != null ?
-            restoreState.getPendingSplits() : Collections.EMPTY_MAP);
+            restoreState.getPendingSplits() : Collections.emptyMap());
```
Comment (zhuangchong) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/enumerator/IcebergBatchSplitEnumerator.java:44]: Why replace that? The two implementations are the same.

**Review by hailin0 — APPROVED** [review]
LGTM

**Review by zhuangchong — APPROVED** [review]
+1

**Review by ashulin — APPROVED** [review]

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**